### PR TITLE
EREGCSC-1619 -- Rollup to Vite

### PR DESCRIPTION
**Pull request by PhilR8**
Resolves [EREGCSC-1619](https://jiraent.cms.gov/browse/EREGCSC-1619)

**Description:**
Update Django-specific front end bundle pipeline to allow using Vuetify etc. This will likely mean migrating from Rollup to Vite.

**This pull request changes:**

- Removes Rollup build pipeline, deleting config file and removing all artifacts and dependencies associated with it.
- Moves former Rollup build pipeline to a Vite build pipeline under the directory `eregs-component-lib`
- Upgrades all Vite instances (`eregs-vite` and `eregs-component-lib`) to `v4`
- Upgrades `Makefile` to use new build pipelines for local development (`make watch` and `make watch-all`) and for deployment
   - `make watch` and `make watch-all` will behave as expected:
      - `make watch` will build and watch only the Django template scripts and styles (so nothing related to the Single Page Apps at `/search` or `/resources`)
      - `make watch-all` will build and watch everything -- Django template scripts and styles, as well as the SPA scripts and styles
- Fixes Storybook issue that was introduced when Node was upgraded to v18

**Steps to manually verify this change:**

1. Pull down repository and make sure everything works as expected on your local machine.
   - `make watch`
   - `make watch-all`
   - `make test`
   - `make storybook`
2. Make sure repository is deploying to experimental deployment without issue and all tests passing.  
3. Poke around and compare experimental deployment to prod site to make sure things match.

### CODE REVIEW

File: solution/backend/cmcs_regulations/settings.py [hunk 1]
```diff
@@ -263,7 +263,15 @@
...
-    "https://cdn.jsdelivr.net/npm/swagger-ui-dist@latest/swagger-ui-standalone-preset.js"
+    "https://cdn.jsdelivr.net/npm/swagger-ui-dist@latest/swagger-ui-standalone-preset.js",
+]
+CSP_SCRIPT_SRC_ELEM = [
+    "'self'",
+    "'unsafe-inline'",
+    "'unsafe-eval'",
+    STATIC_URL,
+    "https://*.googletagmanager.com",
+    "https://cdn.jsdelivr.net/npm/vue@2.7.14",
```
Comment (PhilR8) [inline solution/backend/cmcs_regulations/settings.py:274]: loading vue via CDN for Django template pages

File: solution/backend/regulations/templates/regulations/about.html [hunk 2]
```diff
@@ -256,5 +256,6 @@ <h2>Technical Details</h2>
 {% endblock %}
 
 {% block post_footer %}
-<script src="{% static '/js/main.build.js' %}"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vue@2.7.14"></script>
```
Comment (PhilR8) [inline solution/backend/regulations/templates/regulations/about.html:259]: loading vue via CDN for Django template pages

File: solution/ui/regulations/.storybook/main.js [hunk 3]
```diff
@@ -37,6 +37,13 @@ module.exports = {
             "@": path.resolve(__dirname, "../../prototype/src"),
         };
 
+        config.module.rules.push({
+            test: /\.m?js/,
+            resolve: {
+                fullySpecified: false,
+            },
+        });
+
```
Comment (PhilR8) [inline solution/ui/regulations/.storybook/main.js:46]: see: https://github.com/graphql/graphql-js/issues/2721

File: solution/ui/regulations/package.json [hunk 4]
```diff
@@ -5,15 +5,13 @@
     "css": "npx sass -I node_modules css/scss/main.scss ../../static-assets/regulations/css/main.css",
     "css-deploy": "npx sass -I node_modules css/scss/main-deploy.scss ../../static-assets/regulations/css/main.css",
     "watch-css": "npx sass --watch -I node_modules css/scss/main.scss ../../static-assets/regulations/css/main.css",
-    "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook",
-    "watch-js": "npx rollup -c --watch --sourcemap",
-    "js": "npx rollup -c --sourcemap"
+    "storybook": "NODE_OPTIONS=--openssl-legacy-provider start-storybook -p 6006",
+    "build-storybook": "NODE_OPTIONS=--openssl-legacy-provider build-storybook"
```
Comment (PhilR8) [inline solution/ui/regulations/package.json:9]: see: https://www.newline.co/@kchan/how-to-fix-the-error-errorerror0308010cdigital-envelope-routinesunsupported--0f8d3f17

**Review by thwalker6 — APPROVED** [review]
lgtm

**Pull request merged**